app/lib/url.rb

```diff
@@ -41,8 +41,12 @@ def self.article(article)
   # Creates a comment URL
   #
   # @param comment [Comment] the comment to create the URL for
-  def self.comment(comment)
-    url(comment.path)
+  # @param article_path [String, nil] the path of the article to anchor
+  #   the comment link instead. Return the comment permalink if nil
+  def self.comment(comment, article_path: nil)
+    return url(comment.path) if article_path.nil?
+
+    url("#{article_path}#comment-node-#{comment.id}")
```

Comment:
Just want to call out from the issue that we're looking to have this point to a new element named "comment-`id_code` because the "id code" is different from ID, which can remain as the one used for certain ways that JavaScript relies on it to match with the comment ID. This seems like a point that's easy to miss in the context of the problem 😄 

`id_code` is the human-readable comment reference we want to stick with that maps directly to the current permalink.

Again, not sure if this is a WIP state headed in that direction, but figured I'd comment in case it could save time later.

Reply:
Thanks for pointing this out @benhalpern!

The reason I was using the comment's id instead of `id_code` was that the DOM was already populated with the comment's id instead:

<img width="1055" alt="Screen Shot 2022-09-30 at 13 34 38" src="https://user-images.githubusercontent.com/6045239/193343783-a1b18bb9-d69b-401c-ba95-7b6e27931d9a.png">

I can switch both the DOM element's id and this selector to use that instead. I have to see if that breaks some other functionality though (I'll share that here in a bit)

Reply:
Just FYI to make sure it's clear, when logged in the permalink (before making the change to `id_code`) would continue to use the `id_code` as it works right now in (before this PR). When logged out is when the link would've used the id for the anchor tag like `/username/slug#comment-node-<id>`.

I'll still try to make this change now to maintain the consistency on both situations though.

Reply:
Yes, that is all true.

IMO we should keep `comment-note-#{id}` as it is already in place because its purpose as a JS reference makes sense, with stuff like:

```javascript
var parentComment = document.getElementById("comment-node-" + event.target.dataset.commentId);
```

I think this new reference (and hash references can either refer to an `id` element or a `name` element, and I think `name` is cleaner because `id` is used in so many other contexts) should be crafted so it is _only_ used for this purpose, to remove the need for this attribute to carry too much load.

***

One final thought: It's a small thing, but currently if I visit https://dev.to/jordanpowell88/how-i-became-a-google-developer-expert-4ee0#comment-node-950784 (just as a random example) the sticky navbar touches it _exactly_ at the top of the comment, like this:

<img width="309" alt="Screen Shot 2022-09-30 at 4 29 22 PM" src="https://user-images.githubusercontent.com/3102842/193351507-02b13b5e-ae30-4398-8da3-505510f5d6ae.png">

And I think we really want to leave a bit more room, like this:

<img width="353" alt="Screen Shot 2022-09-30 at 4 29 28 PM" src="https://user-images.githubusercontent.com/3102842/193351560-3000119c-ebc5-4c5c-8fbb-6d4a8a17ec88.png">

I think if we unbound the new `name` attribute, we might be able to address these minor issues with more precision, or respond to a follow up change if necessary.

I might be wrong with any of my assumptions here, so ultimately go ahead and make the most practical choices, but I hope this clarifies anything for you!

Reply:
Oh absolutely, now it's much clearer what you mean by all this, thanks! I believe I can get those changes in this PR 👍🏼